plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/CachingJdbcClient.java

```diff
@@ -554,4 +582,46 @@ public int hashCode()
             throw new UncheckedExecutionException(e);
         }
     }
+
+    private static final class TableStatisticsCacheKey
+    {
+        private final JdbcIdentity identity;
```

Comment:
I see no reason to make identity part of the cache key. Please remove or add code comment.

Reply:
It might the case that each user see different data (row filtering or column masking which is applied in remote data source).

Reply:
Add a comment then

Reply:
for a typical JDBC connector, it's not possible. Also, row filtering and masking is applied on top of TS, so here we shouldn't be concerned about that. I am concerned that identiy here will decrease cache effectiveness (well, this part is obvious). Do we have a specific use-case in mind, or you're trying to stay on the safe side here?

Reply:
> row filtering and masking is applied on top of TS, so here we shouldn't be concerned about that.

It is not for Trino row filtering and column masking, but row filtering and column masking in remote data source.

> Do we have a specific use-case in mind,

https://www.postgresql.org/docs/9.5/ddl-rowsecurity.html

> you're trying to stay on the safe side here?

Yes. I think I am just going crazy here and possibly it is not a practical problem. My biggest fear is that if I disregard this then we might have a security issue (data leak)

Reply:
Same applies to metastore caching. There, we cache for individual users when impersonation is enabled and globally when it is not.